### PR TITLE
Added Histogram Normalization Tutorial

### PR DESCRIPTION
**Pull request by AdvaitDhingra**
# This Pull request:

Adds a tutorial demonstrating the 7 ways to normalize a histogram.

## Changes or fixes:

Above

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

**Comment by AdvaitDhingra** [comment]
The image produced looks a little weird and I don't know how to fix that. I could separate the plots on to 7 TCanvases but idk if that's efficient.

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:70]: @couet do we need this?

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:32]:
```suggestion
/// Double_t scale = h->GetXaxis()->GetBinWidth(1)/h->Integral();
```

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:28]:
```suggestion
///  h->Scale(1./h->GetEntries());
```

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:19]: The bin width cannot really be 0 - the tutorial shouldn't bother to check, and especially not the doc.

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:38]:
```suggestion
/// h->Scale(1./h->Integral());
```

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:43]:
```suggestion
/// h->Scale(1., "width");
```

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:48]:
```suggestion
/// h->Scale(1./h->Integral(), "width");
```

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:53]:
```suggestion
/// h->Scale(1./h->Integral("width"));
```

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:7]: We shouldn't show how many ways there are, but demonstrate just the two main, most simple interfaces:

- with `"width"` argument: `h->Scale(1./h->Integral("width");` - his should be the default
- without `"width"` argument: `h->Scale(1./h->Integral();`

File: tutorials/hist/Normalizing_Histogram.C
Comment (Axel-Naumann) [inline tutorials/hist/Normalizing_Histogram.C:4]: Maybe nicer: "Normalize a Histogram" / `NormalizeHistogram.C`?

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: tutorials/hist/Normalizing_Histogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/Normalizing_Histogram.C:4]: Yeah, that sounds better

File: tutorials/hist/Normalizing_Histogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/Normalizing_Histogram.C:65]: I thought it makes the whole thing cleaner

**Comment by couet** [comment]
I ran your macro you plot 7 times the initial histogram. One is enough.
The titles are too small.

**Comment by AdvaitDhingra** [comment]
> I ran your macro you plot 7 times the initial histogram. One is enough.
> The titles are too small. 

Okay, I'll fix it at ~2pm

**Comment by AdvaitDhingra** [comment]
Is this better? Sorry for the amount of commits, I can squash them before merge.

File: tutorials/hist/Normalizing_Histogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/Normalizing_Histogram.C]: Okay, I will remove the synonymous ones

Comment (AdvaitDhingra) [reply tutorials/hist/Normalizing_Histogram.C]: Okay, I just kept the ones you described above, as the other ones were basically the same.

File: tutorials/hist/Normalizing_Histogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/Normalizing_Histogram.C]: Okay. I removed that method anyway so it shouldn't matter. The only methods are the one with "width" and the one without, so a total of 2.

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:20]: Can you explain the different of the two methods, here?

The first is simply normalizing by dividing by the bin content; the second is also taking the bin width into account, i.e. normalizing by the sum of the bin's content times their respective widths. This is especially important for variable-sized bins, but also for giving a physical meaning to the norm, such as "20.2 events / GeV" for an x-axis unit of GeV.

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:38]: Please use more descriptive / telling / educational names and titles, esp in tutorials. Suggestions:

```suggestion
   TH1F *orig = new TH1F("orig", "Original histogram before normalization", 100, 0, 100);
   orig->SetStats(0);
   TH1F *norm = new TH1F("norm", "Histogram normalized with bin content", 100, 0, 100);
   norm->SetStats(0);
   TH1F *normw = new TH1F("normw", "Histogram normalized with bin content times bin width", 100, 0, 100);
```

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:39]: Why do you have these calls to `SetStats`? I think it might be very useful to demonstrate the effect of normalization on the hist stats.

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:43]: Why heap? This is better:

```suggestion
   TRandom2 rand;
```

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:47]: We generally use / recommend - for iterators this used to be cheaper, so it's a better pattern to follow.
```suggestion
   for (int i = 0; i < 1000; ++i) {
```

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:46]: I'd remove some of the empty lines, e.g. this one.

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:54]: This is where the explanation should go! See above.

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [reply tutorials/hist/NormalizeHistogram.C:20]: Actually - it's better to explain this below, where the normalization actually happens. Here you can just remove the part "the methods are" because it's a repetition of the actual code below.

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:8]:
```suggestion
/// Two different methods of normalizing histograms
/// are shown, each with the original histogram.
```

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:38]: If you use bin width == 1 then there's no difference between the normalizations with and without `"width"` :-) I'd recommend to use:
```suggestion
   std::array<double, 6> binsx{0, 5, 10, 20, 50, 100};
   TH1F *hist1 = new TH1F("Unnormalized Histogram", "Unnormalized Hist", binsx.size() - 1, &binsx[0]);
   hist1->SetStats(0);
   TH1F *hist2 = new TH1F("Method 1 Normalized", "Method 1 Normalized", binsx.size() - 1, &binsx[0]);
   hist2->SetStats(0);
   TH1F *hist3 = new TH1F("Method 2 Normalized", "Method 2 Normalized", binsx.size() - 1, &binsx[0]);
```

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:51]: Instead of filling all three histograms we can fill one and then `Clone` it twice, calling `SetTitle()` on the resulting histogram and normalizing the clone. @couet can you guide @AdvaitDhingra here?

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:62]:
```suggestion
   TCanvas *c1 = new TCanvas("c1", "Histogram Normalization", 1000, 600);
```

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:72]: What's the motivation for this line? I think it's not needed.

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: tutorials/hist/NormalizeHistogram.C
Comment (Axel-Naumann) [inline tutorials/hist/NormalizeHistogram.C:6]:
```suggestion
/// Image produced by `.x NormalizeHistogram.C`
```

File: tutorials/hist/NormalizeHistogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/NormalizeHistogram.C:20]: Okay, I'll remove these lines and put the description where the actual methods are.

File: tutorials/hist/NormalizeHistogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/NormalizeHistogram.C:38]: Yeah, thats better.

File: tutorials/hist/NormalizeHistogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/NormalizeHistogram.C:39]: I thought it would make it look cleaner, but you're right, they are useful.

File: tutorials/hist/NormalizeHistogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/NormalizeHistogram.C:47]: Ah ok. I only knew i++ 😅

File: tutorials/hist/NormalizeHistogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/NormalizeHistogram.C:51]: Ah ok. I'm not familiar with that, so maybe @couet can help 😅

File: tutorials/hist/NormalizeHistogram.C
Comment (AdvaitDhingra) [reply tutorials/hist/NormalizeHistogram.C:72]: No particular reason, I just thought that that way the image to be used on the website can be generated. I'll remove it

**Comment by AdvaitDhingra** [comment]
I think it should be good to go now @Axel-Naumann

**Review by Axel-Naumann — COMMENTED** [review]
LGTM. @couet could you check the appearance / look of the canvas etc? I'll let you do the merge!

**Review by Axel-Naumann — APPROVED** [review]

File: tutorials/hist/NormalizeHistogram.C
Comment (couet) [inline tutorials/hist/NormalizeHistogram.C:26]: 0.06 is better than 0.1

**Comment by couet** [comment]
Fine with to. Only a small text size adjustment.

**Review by couet — APPROVED** [review]

**Review by Axel-Naumann — APPROVED** [review]

**Comment by Axel-Naumann** [comment]
@AdvaitDhingra let me know when this can be merged!

**Comment by AdvaitDhingra** [comment]
@Axel-Naumann I thought everything was ok but then it tells me that these tests fail. Do you know why?

**Comment by Axel-Naumann** [comment]
The failure is unrelated (I need to debug but no time)

**Pull request merged**
